smack-extensions/src/main/java/org/jivesoftware/smackx/muc/MultiUserChatManager.java

```diff
@@ -199,6 +206,40 @@ public void processStanza(Stanza packet) {
         };
         connection.addAsyncStanzaListener(invitationPacketListener, INVITATION_FILTER);
 
+        // Listens for all messages that include an XEP-0249 GroupChatInvitation extension and fire the invitation
+        // listeners
+        StanzaListener directInvitationPacketListener = new StanzaListener() {
+            @Override
+            public void processStanza(Stanza packet) {
+                final Message message = (Message) packet;
+                GroupChatInvitation invite =
+                    packet.getExtension(GroupChatInvitation.class);
+
+                if (invite.getRoomAddress() != null) {
+                    EntityBareJid mucJid = JidCreate.entityBareFromOrNull(invite.getRoomAddress());
+                    if (mucJid == null) {
+                        LOGGER.warning("Received invite to non bare JID: '" + message + "'");
+                        return;
+                    }
+                    // Fire event for invitation listeners
+                    final MultiUserChat muc = getMultiUserChat(mucJid);
+                    final XMPPConnection connection = connection();
+                    final MUCUser.Invite mucInvite = null;
+                    final EntityJid from = message.getFrom().asEntityJidIfPossible();
+                    if (from == null) {
+                        LOGGER.warning("Group Chat Invitation from non entity JID: '" + message.toXML() + "'");
```

Comment:
I am not super happy about that, as I believe it is likely that components, i.e., domain bare JIDs, could also send directed invites. I think we could change the signature of `InvitationsLIstener` to use just `Jid` for `inviter`.

Reply:
I'm not sure I agree. The end user the invitation is from should be the same regardless of if they're using an XEP-0249 direct invitation or a mediated XEP-0045 invitation. If we are asserting a user must be an entity jid in the XEP-0045 case then I think it's very reasonable to assert the same here.